code/modules/holodeck/basketball.dm

```diff
@@ -0,0 +1,107 @@
+//Basketball items and structures
+
+/obj/item/toy/basketball
+	name = "basketball"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "basketball"
+	inhand_icon_state = "basketball"
+	desc = "Here's your chance, do your dance at the Space Jam."
+	w_class = WEIGHT_CLASS_BULKY //Stops people from hiding it in their bags/pockets
+
+/obj/item/toy/basketball/Initialize()
+	. = ..()
+	AddComponent(/datum/component/knockoff, knockoff_chance=30, zone_override=list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM), slots_knockoffable=list(ITEM_SLOT_HANDS))
+
+/obj/structure/hoop
+	name = "basketball hoop"
+	desc = "Boom, shakalaka!"
+	icon = 'icons/obj/basketball.dmi'
+	icon_state = "hoop"
+	anchored = TRUE
+	density = TRUE
+
+/**
+* this proc is for dunking an object
+*
+* ball_exp is how much exp you get for a dunk, ball_skill is balling skill level modifier
+* dunking has a 60% chance of failure which decreases based on balling skill
+*			unsuccessful: 5 exp; 50 stamina damage; ball goes in a random direction one step away
+*			sucessful: 20 exp; 40-10 stamina damage (based on ball_skill); ball goes in net
+*/
+/obj/structure/hoop/attackby(obj/item/dunk_object, mob/user, params)
+	if(get_dist(src,user)>=2) //can only dunk when close
+		return
+
+	var/ball_skill = user.mind?.get_skill_modifier(/datum/skill/balling, SKILL_PROBS_MODIFIER)
```

Comment:
since you have a `?.` check in here it might be worthwhile to nullcheck `ball_skill` before using it a few lines down